xdsl/backend/riscv/lowering/convert_memref_to_riscv.py

```diff
@@ -322,8 +323,7 @@ def match_and_rewrite(self, op: memref.SubviewOp, rewriter: PatternRewriter):
         source = op.source
         result = op.result
         source_type = source.type
-        assert isinstance(source_type, MemRefType)
-        source_type = cast(MemRefType, source_type)
+        assert isa(source_type, MemRefType)
```

Comment:
This looks a bit unrelated

Reply:
This and the other thing below were drive-by fixes, I think they're incremental improvements even if there's more work to do